test/unit/math/prim/mat/vectorize/foo_fun.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef TEST_UNIT_MATH_PRIM_MAT_VECTORIZE_FOO_FUN_HPP
```

Comment:
I don't think so because `exp` is defined everywhere and it wouldn't allow us to test functions that have boundary conditions.